pkg/kubelet/kubelet.go

```diff
@@ -1554,7 +1560,11 @@ func (kl *Kubelet) makeEnvironmentVariables(pod *api.Pod, container *api.Contain
 					return result, err
 				}
 			case envVar.ValueFrom.ResourceFieldRef != nil:
-				runtimeVal, err = containerResourceRuntimeValue(envVar.ValueFrom.ResourceFieldRef, pod, container)
+				defaultedPod, err := kl.defaultPodLimitsForDownwardApi(pod)
+				if err != nil {
+					return result, err
+				}
+				runtimeVal, err = containerResourceRuntimeValue(envVar.ValueFrom.ResourceFieldRef, defaultedPod, container)
```

Comment:
I am not sure how to fix this up... needs some thought.  have no clear way of correlating that container back to a container in the spec.  seems like the defaulting should have happened on the container and not the pod.